### ISSUE TITLE
[v7.8.1] Pipewire doesn't detect default audio output.

### ISSUE THREAD

**Issue by muelsysefoss** [Taiko2k/Tauon#1216]
Heyo, thanks for the sick app, been using it for a while.

I've been playing around with the Pipewire backend on the **Tauon Flatpak** (I'm running **Fedora Kinoite**, so all packages should be updated). Basically, upon launching and attempting to play a song, no sound comes out. This was unusual as flatpak's logs showed that Pipewire was initiated successfully. Then, upon switching the audio device from `Default` to `[My headphones]`, there was audio output. Switched it to my laptop's speakers and there was output there as well. So, detection of default output device is broken somewhere (works fine on the PulseAudio backend).

Not a deal-breaker, it's experimental after all, just wanted to give a heads up. Can someone check on their machine to see if they experience the same issue?

**Comment by C0rn3j** [comment]
Does 7.8.3 flatpak still suffer from the issue?

If so, can you try the latest development build - https://github.com/Taiko2k/Tauon/wiki/Development-builds

If master/devel build does not suffer from the issue, can you try 7.8.3 non-flatpak?

**Comment by muelsyse-az** [comment]
Behavior is stranger in 7.8.3.

While it does find the default output most of the time, sometimes it doesn't. The worst part that it ends up duplicating all the audio sources? I think I'll report that in another issue as this has been technically fixed. Closing this issue is your call.

**Comment by C0rn3j** [comment]
Can you try the 7.9.0 prerelease that we now happen to have?

https://github.com/Taiko2k/Tauon/issues/1084#issuecomment-2571286287

>  I think I'll report that as another issue as this has been technically fixed.

Works "some of the time" sounds a bit buggy still, let's see if the upcoming version fixes it

**Comment by muelsyse-az** [comment]
Right, since I have a little bit of time, gonna give a basic rundown of the peculiarities of Tauon Flatpak's Pipewire support (that persists on v7.9.0).

Tauon, when run under Pipewire, likes to break KDE Plasma's `Audio Volume` System Tray, leading to a `Connection to Sound Service Lost` error. This forces me to open the Sound KCM (KDE Plasma speak for Settings Page) to control my audio settings. According to my logs, it seems this might be related to my USB-C headset but at this moment I'm a little clueless on whether this is a Tauon problem or a Plasma problem or a Pipewire/PulseAudio problem.

![image](https://github.com/user-attachments/assets/39f98426-f9d6-4fe4-a7dd-92e25cd41480)

Despite this error, the default audio output _is_ detected. There have also been instances where the `Set audio output device` list is duplicated when I try and change my audio profiles in order to fix this problem but that's difficult to replicate and I would need more time to see if this is truly fixed with v7.9.0.

**Comment by C0rn3j** [comment]
> Edit: Now that I think about it, why am I getting PulseAudio errors when I'm running through Pipewire?

Because you still need to have a Pulse backend on your system, which is either pulseaudio or pipewire-pulse.  

You should be defaulting to the latter - the fact the plasma applet is named org.kde.pulseaudio is irrelevant to the actual backend you're running.

**Comment by C0rn3j** [comment]
> Tauon, when run under Pipewire, likes to break KDE Plasma's Audio Volume System Tray, leading to a Connection to Sound Service Lost error. This forces me to open the Sound KCM (KDE Plasma speak for Settings Page) to control my audio settings. According to my logs, it seems this might be related to my USB-C headset but at this moment I'm a little clueless on whether this is a Tauon problem or a Plasma problem or a Pipewire/PulseAudio problem.

No such problem here, can you verify it happens with non-Flatpak version too?

![image](https://github.com/user-attachments/assets/9f147041-a1fd-4faf-b611-2630b8c046fc)

Does Fedora keep up with the Plasma/frameworks versions? They tend to have tons of bugfixes that may be relevant in this case.

**Comment by muelsyse-az** [comment]
Sorry for the late reply, I'm still don't have free time this week. Just a quick heads up though.

Even in 7.9.0 Flatpak, Tauon sometimes doesn't detect default audio output. The reason for this seems to be that Tauon connects to this empty Audio node when set to Default. I have no clue why this empty Audio node exists, why Tauon connects to it as Default, and why other applications seem to not have this problem.

![Image](https://github.com/user-attachments/assets/c4f58b9e-3e37-4b33-becf-f6724b605b7b)

**Comment by C0rn3j** [comment]
`pw-cli ls` could possibly tell the culprit, and you should also check pavu/pwvucontrol to see what devices/profiles you have enabled.

**Comment by C0rn3j** [comment]
Do you still have the issue on current versions of software/Tauon?

**Comment by muelsyse-az** [comment]
I can confirm that this issue does not occur anymore on current versions of Tauon. Wonderful stuff.
